File: 03-march/21-flip-square-submatrix-vertically/solution.js
/**
 * @param {number[][]} grid
 * @param {number} x
 * @param {number} y
 * @param {number} k
 * @return {number[][]}
 */
var reverseSubmatrix = function(grid, x, y, k) {
  for (let rowOffset = 0; rowOffset < Math.floor(k / 2); rowOffset++) {
    const top = x + rowOffset;
    const bottom = x + k - 1 - rowOffset;
    for (let col = y; col < y + k; col++) {
      const tmp = grid[top][col];
      grid[top][col] = grid[bottom][col];
      grid[bottom][col] = tmp;
    }
  }
  return grid;
};

File: 03-march/21-flip-square-submatrix-vertically/solution.py
class Solution:
    def reverseSubmatrix(self, grid: list[list[int]], x: int, y: int, k: int) -> list[list[int]]:
        for row_offset in range(k // 2):
            top = x + row_offset
            bottom = x + k - 1 - row_offset
            for col in range(y, y + k):
                grid[top][col], grid[bottom][col] = grid[bottom][col], grid[top][col]
        return grid
